src/modules/commander/state_machine_helper.cpp

```diff
@@ -375,6 +377,11 @@ main_state_transition(const vehicle_status_s &status, const main_state_t new_mai
 
 	if (ret == TRANSITION_CHANGED) {
 		if (internal_state->main_state != new_main_state) {
+			// If transitioning from non-auto to auto state, save the last non-auto state
+			if (!is_auto_state(internal_state->main_state)) {
+				last_non_auto_state = internal_state->main_state;
```

Comment:
If you do two consecutive RTLs this stays in the first non-auto state and will switch back to the manual mode in the first RTL, even if the second RTL was triggered out of an auto state. You need to reset the state in an `else` statement to not have history in your state machine.

Reply:
Let me make sure I understand correctly: If you have the sequence, for example, `Manual -> Mission -> RTL -> Mission -> RTL`, then it will switch back to `Manual` after the final `RTL` completes. Should it not do this? Under what circumstances exactly should it not switch back to a manual mode after an RTL?

Reply:
I guess what @LorenzMeier means is that in this case it should switch back to `Mission` for this case.

This would mean the check should be for any mode that is not RTL, instead of non-auto mode.

Reply:
My expectation would be here that it switches back to the mode it had before RTL, not the last manual mode before RTL - otherwise I could have been through a number of mission - RTL - mission - RTL sequences and be left with the manual mode from an hour ago